torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -84,9 +90,40 @@ const ::gloo::ReductionFunction<T>* reductionFunction(const ReduceOp& r) {
   throw std::runtime_error("Unhandled ReduceOp");
 }
 
+std::vector<cudaStream_t> getStreamVector(AlgorithmEntry& entry) {
+  std::vector<cudaStream_t> streams(entry.streams.size());
+  for (auto i = 0; i < entry.streams.size(); i++) {
+    streams[i] = entry.streams[i].getStream();
+  }
+  return streams;
+}
+
+// synchronizeStreams ensures that the private streams associated with
+// an algorithm entry wait for the public streams to complete, and
+// that the public streams wait for the private streams to complete.
```

Comment:
This function only inserts public -> private synchronization, so the comment is misleading.

Reply:
Thanks for catching this. This was my original intent, but proved to be impossible, so changed it later.